sktime/utils/_testing/estimator_checks.py

```diff
@@ -426,3 +429,31 @@ def check_persistence_via_pickle(Estimator):
     for method in results:
         unpickled_result = getattr(unpickled_estimator, method)(*args[method])
         _assert_almost_equal(results[method], unpickled_result)
+
+
+def check_multiprocessing_determinism(Estimator):
+    if 'n_jobs' in signature(Estimator.__init__).parameters:
+        estimator = _construct_instance(Estimator)
+        fit_args = _make_args(estimator, "fit")
+        ops = ["predict", "predict_proba", "transform"]
+        for op in ops:
+            if hasattr(estimator, op):
+                X_test = _make_args(estimator, op)[0]
+                result_set = []
+                for n_jobs in [1, 4]:
+                    estimator = _construct_instance(Estimator, n_jobs=n_jobs)
```

Comment:
I think I'd prefer to call `estimator.set_params(*{"n_jobs": n_jobs})` here rather than changing the `_construct_instance` method if that works? This way we wouldn't have to reconstruct the instance but could simply set the new params and would have to do all of this only if the estimator has the n_jobs attribute. What do you think?

Reply:
Yes, that makes sense. Switched to using `set_params` and removed changes in `_construct_instance`.